bundles/org.openhab.binding.coolmasternet/src/main/java/org/openhab/binding/coolmasternet/internal/ControllerHandler.java

```diff
@@ -145,26 +150,20 @@ public boolean isConnected() {
 
                 logger.trace("Sending command '{}'", command);
                 OutputStream out = localSocket.getOutputStream();
-                out.write(command.getBytes());
-                out.write("\r\n".getBytes());
+                out.write(command.getBytes(StandardCharsets.US_ASCII));
+                out.write(LF);
 
-                try (Reader isr = new InputStreamReader(localSocket.getInputStream());
-                        BufferedReader in = new BufferedReader(isr)) {
-                    while (true) {
-                        String line = in.readLine();
-                        logger.trace("Read result '{}'", line);
-                        if ("OK".equals(line)) {
-                            return response.toString();
-                        }
-                        response.append(line);
-                        if (response.length() > 100) {
-                            /*
-                             * Usually this loop times out on errors, but in the case that we just keep getting
-                             * data we should also fail with an error.
-                             */
-                            throw new CoolMasterClientError(
-                                    String.format("Got gibberish response to command %s", command));
-                        }
+                final Reader isr = new InputStreamReader(localSocket.getInputStream());
```

Comment:
Shouldn't a charset be specified for this reader?

Reply:
It has been this way since 2 January 2017. I did not write it.

Reply:
Anyhow I modified it to be explicit.